byterun/caml/mlvalues.h

```diff
@@ -214,8 +228,8 @@ typedef opcode_t * code_t;
 
 /* Another special case: objects */
 #define Object_tag 248
-#define Class_val(val) Field((val), 0)
-#define Oid_val(val) Long_val(Field((val), 1))
+#define Class_val(val) Field_imm((val), 0)
+#define Oid_val(val) Long_val(Field_imm((val), 1))
```

Comment:
Internally, the OO id is mutated, but I don't think this is ever visible externally?